libs/api/bpf_syscall.cpp

```diff
@@ -6,188 +6,229 @@
 #include <windows.h>
 #include <WinError.h>
 #include <linux/bpf.h>
+#include <stdexcept>
 
-#define CHECK_SIZE(last_field_name)                                                    \
-    if (!tail_is_zero(                                                                 \
-            attr,                                                                      \
-            offsetof(union bpf_attr, last_field_name) + sizeof(attr->last_field_name), \
-            sizeof(union bpf_attr))) {                                                 \
-        return -EINVAL;                                                                \
-    }
-
-static bool
-tail_is_zero(const void* buf, unsigned int start, unsigned int end)
+template <typename T> class ExtensibleStruct
 {
-    const unsigned char* p = (const unsigned char*)buf + start;
-    const unsigned char* e = (const unsigned char*)buf + end;
+  private:
+    void* _orig;
```

Comment:
Is there some reason the private member names and function argument names have to match something else?
The [coding conventions](https://github.com/microsoft/ebpf-for-windows/blob/main/docs/DevelopmentGuide.md#coding-conventions) for this project state:
> **DON'T** use abbreviations unless they are already well-known terms known by users (e.g., "app", "info"), or are already required for use by developers (e.g., "min", "max", "args"). Examples of bad use would be num_widgets instead of widget_count, and opt_widgets instead of option_widgets or optional_widgets.

Reply:
Tried to make these better.